pybamm/models/submodels/interface/sei/sei_growth.py

```diff
@@ -149,45 +191,75 @@ def get_coupled_variables(self, variables):
-            j_inner = variables["Inner SEI interfacial current density"]
-            j_outer = variables["Outer SEI interfacial current density"]
+            L_inner = variables[f"Inner {self.reaction} thickness"]
+            L_outer = variables[f"Outer {self.reaction} thickness"]
+            j_inner = variables[f"Inner {self.reaction} interfacial current density"]
+            j_outer = variables[f"Outer {self.reaction} interfacial current density"]
             if self.reaction_loc == "interface":
                 a = 1
             else:
                 a = variables["Negative electrode surface area to volume ratio"]
 
+        # Get variables specific to cracks
+        if self.reaction == "SEI on cracks":
+            if self.reaction_loc == "x-average":
+                l_cr = variables["X-averaged negative particle crack length"]
+                dl_cr = variables["X-averaged negative particle cracking rate"]
+            else:
+                l_cr = variables["Negative particle crack length"]
+                dl_cr = variables["Negative particle cracking rate"]
+            spreading_outer = dl_cr / l_cr * (self.param.L_outer_0 / 10000 - L_outer)
+            spreading_inner = dl_cr / l_cr * (self.param.L_inner_0 / 10000 - L_inner)
+        else:
+            spreading_outer = 0
+            spreading_inner = 0
+
         Gamma_SEI = self.param.Gamma_SEI
 
         if self.options["SEI"] == "ec reaction limited":
-            self.rhs = {L_outer: -Gamma_SEI * a * j_outer / 2}
+            self.rhs = {L_outer: -Gamma_SEI * a * j_outer / 2 + spreading_outer}
         else:
             v_bar = self.param.v_bar
             self.rhs = {
-                L_inner: -Gamma_SEI * a * j_inner,
-                L_outer: -v_bar * Gamma_SEI * a * j_outer,
+                L_inner: -Gamma_SEI * a * j_inner + spreading_inner,
+                L_outer: -v_bar * Gamma_SEI * a * j_outer + spreading_outer,
             }
 
     def set_initial_conditions(self, variables):
         if self.reaction_loc == "x-average":
-            L_inner = variables["X-averaged inner SEI thickness"]
-            L_outer = variables["X-averaged outer SEI thickness"]
+            L_inner = variables[f"X-averaged inner {self.reaction} thickness"]
+            L_outer = variables[f"X-averaged outer {self.reaction} thickness"]
         else:
-            L_inner = variables["Inner SEI thickness"]
-            L_outer = variables["Outer SEI thickness"]
+            L_inner = variables[f"Inner {self.reaction} thickness"]
+            L_outer = variables[f"Outer {self.reaction} thickness"]
 
         L_inner_0 = self.param.L_inner_0
         L_outer_0 = self.param.L_outer_0
+
         if self.options["SEI"] == "ec reaction limited":
-            self.initial_conditions = {L_outer: L_inner_0 + L_outer_0}
+            if self.reaction == "SEI on cracks":
+                self.initial_conditions = {L_outer: (L_inner_0 + L_outer_0) / 10000}
```

Comment:
why hard-coded to `10000`?

Reply:
That is what was done in the paper. We had to choose an initial condition that was as close to zero to get the physics right, but didn't cause a division by zero error.